prov/efa/src/rxr/rxr_cq.c

```diff
@@ -391,8 +391,9 @@ int rxr_cq_handle_cq_error(struct rxr_ep *ep, ssize_t err)
 		rxr_ep_dec_tx_pending(ep, peer, 1);
 	if (RXR_GET_X_ENTRY_TYPE(pkt_entry) == RXR_TX_ENTRY) {
 		tx_entry = (struct rxr_tx_entry *)pkt_entry->x_entry;
-		if (err_entry.err != -FI_EAGAIN ||
-		    rxr_ep_domain(ep)->resource_mgmt == FI_RM_ENABLED) {
+		if (err_entry.prov_errno != IBV_WC_RNR_RETRY_EXC_ERR ||
+		    ep->use_zcpy_rx ||
```

Comment:
Not sure this is the right fix. We should return RNR errors based on whether the app asked for the provider to handle resource management. There may be a scenario where an app will use their own receive buffers but wants the provider to handle the retry/backoff logic.

Unfortunately the check on line 396 isn't that check, that's checking whether the "core" provider has resource management enabled. For EFA that is always false. I think I asked for that to be removed in an internal review.